vllm/model_executor/layers/quantization/kernels/MPLinearKernel.py

```diff
@@ -0,0 +1,76 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Callable, Optional, Tuple
+
+import torch
+
+from vllm.model_executor.layers.quantization.utils import replace_parameter
+from vllm.scalar_type import ScalarType
+
+
+@dataclass
+class MPLinearLayerConfig:
+    full_weight_shape: Tuple[int, int]  # [in, out]
+    partition_weight_shape: Tuple[int, int]
+    weight_type: ScalarType
+    act_type: torch.dtype
+    group_size: int
+    zero_points: bool
+    has_g_idx: bool
+
+
+class MPLinearKernel(ABC):
```

Comment:
I love the start of the refactor with this class. My only nit is that "MP" already reads as "multi-processing" inside vLLM, so it takes a second to realize this is "mixed-precision"

Reply:
fair, I think its ok for now though given that this will hopefully be refactored in the near future (i.e. the [WIP Linear Layer Kernel Backends](https://docs.google.com/document/d/1AfgGfF73H_hcXfw6ehYO_l1vHEItsopbxFoV1PvnGIQ/edit?usp=sharing))